pkg/config/validation/validation.go

```diff
@@ -1377,7 +1377,18 @@ func validateTrafficPolicy(policy *networking.TrafficPolicy) Validation {
 		validateLoadBalancer(policy.LoadBalancer, policy.OutlierDetection),
 		validateTLS(policy.Tls),
 		validatePortTrafficPolicies(policy.PortLevelSettings),
-		validateTunnelSettings(policy.Tunnel))
+		validateTunnelSettings(policy.Tunnel),
+		validateProxyProtocol(policy.ProxyProtocol))
+}
+
+func validateProxyProtocol(proxyProtocol *networking.TrafficPolicy_ProxyProtocol) (errs error) {
```

Comment:
can we reject if `tunnel` is also set?

Even if it works in envoy in theory it seems crazy to have PROXY over CONNECT over TLS or something. Best to keep it restricted

Reply:
PROXY should always be outside I believe, but it's plausible to combine it with TLS or CONNECT actually, since that's how some LBs do it (ALB?)

Reply:
I think the implementation here would put CONNECT outside PROXY outside TLS, which doesn't align with what you said above?

TLS + PROXY I am fine with (PROXY outside, as done here), just concerned about CONNECT

Reply:
Good, to make it clear, i will remove proxy protocol outside tunnel